tests/manifests/test_configs_consistency.py

```diff
@@ -0,0 +1,709 @@
+    ):
+        paths_which_do_not_match = []
+        for path_consumer in self.paths_consumers:
+            for path in path_consumer.get_all_paths_in_content(skip_path_consistency_for_files):
+                for parent_mount, mount_node in (
+                    mounted
+                    for mounted_path in self.sources_of_mounted_paths
+                    for mounted in mounted_path.get_mounted_paths()
+                ):
+                    if path.startswith(node_path(parent_mount, mount_node)):
+                        break
+                else:
+                    if path not in ignore_paths_mismatches.get(self.name, []):
+                        paths_which_do_not_match.append(path)
+        assert paths_which_do_not_match == [], (
+            f"Paths which do not match an actual file in {self.template_id}/{self.name}: {paths_which_do_not_match}. "
+            f"Skipped {skip_path_consistency_for_files}\n"
+            f"Looked in {self.sources_of_mounted_paths}\n"
+        )
+
+    def mutate_empty_dirs(self, container_spec, workload_spec):
+        for consumer in self.paths_consumers:
+            consumer.mutate_empty_dirs(container_spec, workload_spec, self.mutable_empty_dirs)
+
+
+def traverse_containers_and_run_consistency_check(
+    templates, other_secrets, validated_container_class, consistency_check
+):
+    workloads = [t for t in templates if t["kind"] in ("Deployment", "StatefulSet", "Job")]
+    for template in workloads:
+        # A list of empty dirs that will be updated as we traverse containers
+        all_workload_empty_dirs = {}
+        # Gather all containers and initContainers from the template spec
+        workload_spec = template["spec"]["template"]["spec"]
+        containers = workload_spec.get("initContainers", []) + template["spec"]["template"]["spec"].get(
+            "containers", []
+        )
+        weight = None
+        if "pre-install,pre-upgrade" in template["metadata"].get("annotations", {}).get("helm.sh/hook", ""):
+            weight = int(template["metadata"]["annotations"].get("helm.sh/hook-weight", 0))
+
+        for container_spec in containers:
+            deployable_details = template_to_deployable_details(template, container_spec["name"])
+            validated_container_config = validated_container_class.from_container_spec(
+                template_id(template),
+                workload_spec,
+                container_spec,
+                weight,
+                deployable_details,
+                templates,
+                other_secrets,
+                all_workload_empty_dirs,
+            )
+            consistency_check(
+                validated_container_config,
+                deployable_details.skip_path_consistency_for_files,
+                deployable_details.ignore_unreferenced_mounts,
+                deployable_details.ignore_paths_mismatches,
+            )
```

Comment:
Rather than passing all these in, `ValidatedContainerConfig.from_container_spec` takes a `DeployableDetails`. It should save that and each test uses that as appropriate